snap/squashfs/squashfs.go

```diff
@@ -102,6 +102,32 @@ func (s *Snap) ReadFile(path string) (content []byte, err error) {
 	return ioutil.ReadFile(filepath.Join(unpackDir, path))
 }
 
+// ListDir returns the content of a single directory inside a squashfs snap.
+func (s *Snap) ListDir(path string) ([]string, error) {
+	tmpdir, err := ioutil.TempDir("", "read-dir")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	unpackDir := filepath.Join(tmpdir, "unpack")
+	if err := runCommand("unsquashfs", "-i", "-d", unpackDir, s.path, path); err != nil {
```

Comment:
If all we need are the names we can get away without unpacking (which is of course expensive) and use `unsquashfs -np-progress -dest "" -ls $snapfile.snap $path` (it writes a ugly header, so its probably best to skip the output until the first line with `^/`, we should just add a flag to squashfs-tools and/or make this header go to stderr instead of stdout or something).